src/main/objects/bass/BassTranslator.js

```diff
@@ -493,19 +646,110 @@ function BassTranslator() {
         return [elements, this.makeChoiceAndSplit(functions)]
     }
 
+    this.removeRevolutionFromExtra = function (argsMap, revolution) {
+        for (var a = 0; a < argsMap.extra.length; a++) {
+            if (argsMap.extra[a][0] === (revolution + "")) {
+                argsMap.extra.splice(a, 1)
+            }
+        }
+    }
+
+    this.increaseByHalfTone = function (a) {
+        if (a === undefined) {
+            return undefined
+        }
 
-    this.handleAlterations = function (harmonicFunctions, chordElements, mode, meter, durations) {
+        if (a.length === 1) {
+            return a + "<"
+        }
+
+        if (a.length === 2 && a[1] === ">") {
+            return a[0]
+        } else {
+            return a + "<"
+        }
+    }
+
+    this.handleDownChord = function (argsMap) {
+        if (Utils.contains(argsMap.extra, "1>")) {
+            argsMap.omit = []
+            argsMap.mode = Consts.MODE.MINOR
+            argsMap.down = true
+            var extra = []
+            for (var a = 0; a < argsMap.extra.length; a++) {
+                if (parseInt(argsMap.extra[a][0]) > 5) {
+                    extra.push(this.increaseByHalfTone(argsMap.extra[a]))
+                }
+            }
+            argsMap.extra = extra
+            if (argsMap.position !== undefined) {
+                argsMap.position = this.increaseByHalfTone(argsMap.position)
+            }
+            if (argsMap.revolution !== undefined) {
+                argsMap.revolution = this.increaseByHalfTone(argsMap.revolution)
+            }
+            //probably?
+            for (var a = 0; a < argsMap.delay.length; a++) {
+                for (var b = 0; b < argsMap.delay[a].length; b++) {
+                    argsMap.delay[a][b] = this.increaseByHalfTone(argsMap.delay[a][b])
+                }
+            }
+        }
+    }
+
+
+    this.fixExtraAfterModeChange = function (argsMap) {
+        if (argsMap.mode === Consts.MODE.MAJOR && Utils.contains(argsMap.extra, "3")) {
+            argsMap.extra.splice(argsMap.extra.indexOf("3"), 1)
+        }
+        if (argsMap.mode === Consts.MODE.MINOR && Utils.contains(argsMap.extra, "3>")) {
+            argsMap.extra.splice(argsMap.extra.indexOf("3>"), 1)
+        }
+    }
+
+    this.handleThirdAlterationIn236Chords = function (argsMap, toOmit, toExtra) {
+        if (Utils.contains([2, 3, 6], argsMap.degree)) {
+            if (argsMap.mode === Consts.MODE.MAJOR || argsMap.degree === 2) {
+                if (Utils.contains(toOmit, 3) && Utils.contains(toExtra, "3<")) {
+                    toOmit.splice(toOmit.indexOf(3), 1)
+                    toOmit.push("3>")
+                    toExtra.splice(toExtra.indexOf("3<"), 1)
+                    toExtra.push("3")
+                }
+            } else {
+                //nothing?
+            }
+        }
+    }
+
+    this.addOmit3ForS2IfNecessary = function(argsMap) {
```

Comment:
To niezbyt ładne, powinno być bardziej generyczne

Reply:
Mógłbym później, w ramach refactoru spróbować to uporządkować i wrzucić ogarnianie tego gdzieś wyżej